validator/src/journal/chain_head_lock_ffi.rs

```diff
@@ -0,0 +1,95 @@
+use std::os::raw::c_void;
+
+use cpython::Python;
+use py_ffi;
+
+use journal::chain_head_lock::{ChainHeadGuard, ChainHeadLock};
+use journal::publisher_ffi::convert_on_chain_updated_args;
+
+#[repr(u32)]
+#[derive(Debug)]
+pub enum ErrorCode {
+    Success = 0,
+    NullPointerProvided = 0x01,
+}
+
+#[no_mangle]
+pub extern "C" fn chain_head_lock_lock(
```

Comment:
This function should stay consistent with the C library function naming scheme, it should be renamed `chain_head_lock_acquire()` to match the method name.